src/librustdoc/fold.rs

```diff
@@ -18,30 +18,15 @@ crate trait DocFolder: Sized {
             StrippedItem(..) => unreachable!(),
             ModuleItem(i) => ModuleItem(self.fold_mod(i)),
             StructItem(mut i) => {
-                let num_fields = i.fields.len();
                 i.fields = i.fields.into_iter().filter_map(|x| self.fold_item(x)).collect();
-                if !i.fields_stripped {
-                    i.fields_stripped =
-                        num_fields != i.fields.len() || i.fields.iter().any(|f| f.is_stripped());
```

Comment:
It seems the new on-demand logic doesn't account for fields being removed rather than just being marked stripped. Is it actually possible for fields to just be removed altogether during folding? If so, we might need to change that for this PR to land.

Reply:
Actually no, they were removed from the `strip_hidden` pass.